pkgs/cupertino_http/example/integration_test/url_session_task_test.dart

```diff
@@ -235,22 +235,30 @@ void testURLSessionTaskCommon(
       task.toString(); // Just verify that there is no crash.
     });
 
-    test('cancel', () {
-      task.cancel();
-      if (suspendedAfterCancel) {
-        expect(
-          task.state,
-          NSURLSessionTaskState.NSURLSessionTaskStateSuspended,
-        );
-      } else {
-        expect(
-          task.state,
-          NSURLSessionTaskState.NSURLSessionTaskStateCanceling,
-        );
-      }
-      expect(task.response, null);
-      task.toString(); // Just verify that there is no crash.
-    });
+    test(
+      'cancel',
+      () {
+        task.cancel();
+        if (suspendedAfterCancel) {
+          expect(
+            task.state,
+            NSURLSessionTaskState.NSURLSessionTaskStateSuspended,
+          );
+        } else {
+          expect(
+            task.state,
+            NSURLSessionTaskState.NSURLSessionTaskStateCanceling,
+          );
+        }
+        expect(task.response, null);
+        task.toString(); // Just verify that there is no crash.
+      },
+      // After the request cancellation is complete, the task state will be
+      // `NSURLSessionTaskStateCompleted`. So run the test a few times to allow
```

Comment:
Why do we care about testing this specific behavior if it's so hard to observe it in usage? Should we allow the completed case as a successful test?

```dart
        expect(
          task.state,
          anyOf(
            NSURLSessionTaskState.NSURLSessionTaskStateSuspended,
            NSURLSessionTaskState.NSURLSessionTaskStateCompleted,
          ),
        );
```

Reply:
> Why do we care about testing this specific behavior if it's so hard to observe it in usage?

At some point my `.toString()` would throw for certain states so I created a test to catch that. But it might not be worth it.

> Should we allow the completed case as a successful test?
> 
> ```dart
>         expect(
>           task.state,
>           anyOf(
>             NSURLSessionTaskState.NSURLSessionTaskStateSuspended,
>             NSURLSessionTaskState.NSURLSessionTaskStateCompleted,
>           ),
>         );
> ```

That would work but it would miss the intent of the test i.e. to verify that a cancelled test is actually cancelled instead of running to completion (and to verify that `toString` works correctly ;-))